src/main/java/net/pms/dlna/ImageInfo.java

```diff
@@ -0,0 +1,621 @@
+		this.imageIOSupport = false;
+	}
+
+	/**
+	 * Tries to parse Exif orientation from the given metadata. If it fails in
+	 * in any way, {@code 0} (Normal) is returned.
+	 *
+	 * @param metadata the {@link Metadata} to parse.
+	 * @return The Exif orientation value or {@code 0}.
+	 */
+	public static int parseExifOrientation(Metadata metadata) {
+		if (metadata == null) {
+			return 0;
+		}
+		try {
+			for (Directory directory : metadata.getDirectories()) {
+				if (directory instanceof ExifIFD0Directory) {
+					if (((ExifIFD0Directory) directory).containsTag(ExifIFD0Directory.TAG_ORIENTATION)) {
+						return ((ExifIFD0Directory) directory).getInt(ExifIFD0Directory.TAG_ORIENTATION);
+					}
+				}
+			}
+		} catch (MetadataException e) {
+			return 0;
+		}
+		return 0;
+	}
+
+	/**
+	 * @return The image width or -1 if unknown.
+	 */
+	public int getWidth() {
+		return width;
+	}
+
+	/**
+	 * @return The image height or -1 if unknown.
+	 */
+	public int getHeight() {
+		return height;
+	}
+
+	/**
+	 * @return The {@link ImageFormat} or {@code null} if unknown.
+	 */
+	public ImageFormat getFormat() {
+		return format;
+	}
+
+	/**
+	 * @return The image size in bytes or {@link #SIZE_UNKNOWN}
+	 *         if the size is unknown.
+	 */
+	public long getSize() {
+		return size;
+	}
+
+	/**
+	 * @return The {@link ColorSpace} or {@code null} if unknown.
+	 */
+	public ColorSpace getColorSpace() {
+		return colorSpace;
+	}
+
+	/**
+	 * @return The {@link ColorSpaceType} or {@code null} if unknown.
+	 */
+	public ColorSpaceType getColorSpaceType() {
+		if (colorSpace != null) {
+			return ColorSpaceType.toColorSpaceType(colorSpace.getType());
+		}
+		return colorSpaceType;
+	}
+
+	/**
+	 * @return The number of bits per pixel or -1 if unknown.
+	 */
+	public int getBitsPerPixel() {
+		return bitDepth < 0 || numComponents < 0 ? -1 : bitDepth * numComponents;
+	}
+
+	/**
+	 * The number of components describe how many "channels" the color model
+	 * has. A grayscale image without alpha has 1, a RGB image without alpha
+	 * has 3, a RGP image with alpha has 4 etc.
+	 *
+	 * @return The number of components in the {@link ColorModel} or -1 if
+	 *         unknown.
+	 */
+	public int getNumComponents() {
+		return numComponents;
+	}
+
+	/**
+	 * @return The number of bits per color "channel" or -1 if unknown.
+	 *
+	 * @see #getBitPerPixel()
+	 * @see #getNumColorComponents()
+	 */
+	public int getBitDepth() {
+		return bitDepth;
+	}
+
+	/**
+	 * @return The {@link Metadata} or {@code null} if unknown.
+	 */
+	public Metadata getMetadata() {
+		return metadata;
+	}
+
+	/**
+	 * @return The Exif orientation if any, or {@code 0} if unknown.
+	 */
+	public int getExifOrientation() {
+		return exifOrientation;
+	}
+
+	/**
+	 * @return Whether or not {@link ImageIO} can read/parse this image.
+	 */
+	public boolean isImageIOSupported() {
+		return imageIOSupport;
+	}
+
+	/**
+	 * {@link Metadata} isn't {@link Cloneable} or otherwise easily copied,
```

Comment:
It's a link to [java.lang.Cloneable](https://docs.oracle.com/javase/7/docs/api/java/lang/Cloneable.html) so I guess I'll have to stick to their spelling... 😉